src/hypothesis/database.py

```diff
@@ -20,19 +20,48 @@
 import os
 import re
 import binascii
+import warnings
 import threading
 from hashlib import sha1
 from contextlib import contextmanager
 
+from hypothesis.errors import HypothesisWarning
 from hypothesis._settings import note_deprecation
+from hypothesis.configuration import storage_directory
 from hypothesis.internal.compat import FileNotFoundError, hbytes, \
     b64decode, b64encode
+from hypothesis.utils.conventions import infer
 
 sqlite3 = None
 SQLITE_PATH = re.compile(r"\.\(db|sqlite|sqlite3\)$")
 
 
 def _db_for_path(path=None):
+    if path is infer:
```

Comment:
The reuse of `infer` to mean something different from type inference is a bit weird. This doesn't seem to be documented anywhere. Why is it different from `not_set`?

Reply:
Whoops, too used to reaching for the other constant.  I'll push a fixed version as soon as `master` updates.